### PR TITLE
Fix: Parse cluster_by in dbt models before converting to SQLMesh models

### PR DESCRIPTION
**Pull request by izeigerman**

### CODE REVIEW

File: sqlmesh/dbt/model.py [hunk 1]
```diff
@@ -226,7 +226,9 @@ def to_sqlmesh(self, context: DbtContext) -> Model:
             )
 
         if self.cluster_by:
-            optional_kwargs["clustered_by"] = self.cluster_by
+            optional_kwargs["clustered_by"] = [
+                d.parse_one(c, dialect=dialect).name for c in self.cluster_by
+            ]
```
Comment (crericha) [inline sqlmesh/dbt/model.py:231]: Do we need the parser or can we instead do `exp.to_column(c).name`?

Comment (crericha) [reply sqlmesh/dbt/model.py:231]: Also, do we properly quote CLUSTER BY cols in the generated sql?

Comment (izeigerman) [reply sqlmesh/dbt/model.py:231]: Nope, need to parse it to get rid of quotes:
```
In [5]: exp.to_column('`col`').name
Out[5]: '`col`'
```

Comment (crericha) [reply sqlmesh/dbt/model.py:231]: There is probably a sqlglot way to get the non-quoted text. @tobymao would know.

Comment (tobymao) [reply sqlmesh/dbt/model.py:231]: you need to pass in the dialect, to_column

**Pull request merged**